### ISSUE TITLE
doesSessionExists should check the front-token validity and refresh if necessary

### ISSUE THREAD

**Issue by porcellus** [supertokens/supertokens-website#175]
Since we are not checking the front-token validity, a refresh call may be necessary even if `doesSessionExists` returns true. 
This means that future calls to things like `getAccessTokenPayloadSecurely` may throw if the refresh call throws.

**Comment by leonard-henriquez** [comment]
Since I raised this issue and I know how to reproduce it, I would be delighted to test the fix to this issue when you have one !

**Comment by rishabhpoddar** [comment]
hey @leonard-henriquez 

We have a fix for the problem, but it would be great if you can test it out in your setup and see if it solves the issue.

Can you run the following command on your project to get the new (unreleased) changes:
```bash
cd your_project
npm i git+https://github.com:supertokens/supertokens-website#fix/doesSessionExist_refresh
```

Rebuild your app and try again. To reproduce the issue, do the following:
- Set the access token lifetime to 15 seconds on the supertokens.com dashboard
- Login to your app and wait for 15 seconds
- Block the refresh endpoint
- Reload the page. 

If the issue is fixed, you should now see the login screen. 

Then unblock the refresh endpoint and reload the page again, and you should see the dashboard screen.

Thanks!

**Comment by rishabhpoddar** [comment]
@leonard-henriquez did you get a chance to try the fix please? If not, that's OK, we will be looking to release this within the next 2 days anyway.

**Comment by leonard-henriquez** [comment]
Sorry for the late response @rishabhpoddar , but I tired it and it works 🎉 !!

**Comment by rishabhpoddar** [comment]
This has been fixed in version >= 14.0.3
